internal/webhooks/clusterclass.go

```diff
@@ -246,6 +250,32 @@ func validateUpdatesToMachineHealthCheckClasses(clusters []clusterv1.Cluster, ol
 	return allErrs
 }
 
+func (webhook *ClusterClass) validateKubernetesVersionsOfClusters(clusters []clusterv1.Cluster, _, newClusterClass *clusterv1.ClusterClass) field.ErrorList {
+	var allErrs field.ErrorList
+
+	// If there is no KubernetesVersions is set in the ClusterClass return early.
+	if len(newClusterClass.Spec.KubernetesVersions) == 0 {
```

Comment:
Asking my question regarding this point here
>ClusterClass webhook: on create/update: ensure versions of all Clusters using that ClusterClass are included in ClusterClass versions **_(if CC has versions)_**

I am bit confused about this check *_(if CC has versions)_**

For example we have a CC A with versions 1.32, 1.33, Then we create new CC B for rebase with no version. Since there are no KubernetesVersion set on CC B, this check makes it as valid.
But in this case what will the Cluster use, will it continue to remain in its version?

Reply:
Got it, Thanks for clarifying.